api/hypershift/v1beta1/azure.go

```diff
@@ -480,16 +480,55 @@ type AzureResourceManagedIdentities struct {
 	DataPlane DataPlaneManagedIdentities `json:"dataPlane"`
 }
 
+// AzureClientID is a string that represents the client ID of a managed identity.
+//
+// +kubebuilder:validation:XValidation:rule="self.matches('^[{]?[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}[}]?$')",message="the client ID of a managed identity must be a valid UUID. It should be 5 groups of hyphen separated hexadecimal characters in the form 8-4-4-4-12."
+// +kubebuilder:validation:MaxLength=255
```

Comment:
This length isn't correct is it? Don't these IDs have a fixed length? Also, probably worth setting a minimum length too, IIUC, in this case, it would be the same value for minimum and maximum?

Reply:
Yeah that makes sense